src/append_only_data.rs

```diff
@@ -1026,4 +1045,70 @@ mod tests {
         let decoded = unwrap!(self::Address::decode_from_zbase32(&encoded));
         assert_eq!(address, decoded);
     }
+
+    #[test]
+    fn append_unseq_data_test() {
+        let mut data = UnpubUnseqAppendOnlyData::new(XorName(rand::random()), 10);
+
+        // Assert that the current Append Operation has duplicate entries present and are not pushed
+        let entries = vec![
+            (b"KEY1".to_vec(), b"VALUE1".to_vec()),
+            (b"KEY2".to_vec(), b"VALUE2".to_vec()),
+            (b"KEY1".to_vec(), b"VALUE1".to_vec()),
+        ];
+        assert_eq!(Error::DuplicateEntryKeys, unwrap_err!(data.append(entries)));
+
+        // Assert that the current Append Operation has no duplicate entries present and are pushed
```

Comment:
Maybe rephrase it a bit? Also, we're looking for duplicate keys as opposed to entries (keys + values).
```suggestion
        // Assert that the entries are appended because there are no duplicate keys.
```